### PR TITLE
Types 2.0

### PR DESCRIPTION
**Pull request by aliib**
Please fill in this template.

- [x] Prefer to make your PR against the `types-2.0` branch.
- [x] The package does not provide its own types, and you can not add them.
- [x] Test the change in your own code.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).

If changing an existing definition:
- [x] Provide a URL to  documentation or source code which provides context for the suggested changes: http://electron.atom.io/docs/api/net/
- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by yuit** [comment]
@aliib  Thank you for your contribution. The build break is fix in this PR #12594 . I will let you know once that merge and you should update your PR

**Comment by andy-ms** [comment]
@aliib Please merge the latest `types-2.0` to fix tests.

File: electron/index.d.ts [hunk 1]
```diff
@@ -5565,6 +5806,7 @@ declare namespace Electron {
 		globalShortcut: Electron.GlobalShortcut;
 		Menu: typeof Electron.Menu;
 		MenuItem: typeof Electron.MenuItem;
+        net: Electron.Net;
```
Comment (rhysd) [inline electron/index.d.ts]: nit: Indentation is wrong.

File: electron/index.d.ts [hunk 2]
```diff
@@ -2687,6 +2687,247 @@ declare namespace Electron {
...
+	 */
+	interface Net extends NodeJS.EventEmitter {
+		/**
+		 * @param options The ClientRequest constructor options.
+		 * @param callback A one time listener for the response event.
+		 *
+		 * @returns a ClientRequest instance using the provided options which are directly
+		 * forwarded to the ClientRequest constructor.
+		 */
+		request(options : string | RequestOptions, callback?: (response: IncomingMessage) => void): ClientRequest
```
Comment (rhysd) [inline electron/index.d.ts]: nit: semicolon is missing

File: electron/index.d.ts [hunk 3]
```diff
@@ -2687,6 +2687,247 @@ declare namespace Electron {
...
+	 * and monitoring.
+	 *
+	 * The net module API has been specifically designed to mimic, as closely as possible,
+	 * the familiar Node.js API. The API components including classes, methods,
+	 * properties and event names are similar to those commonly used in Node.js.
+	 *
+	 * The net API can be used only after the application emits the ready event.
+	 * Trying to use the module before the ready event will throw an error.
+	 */
+	interface Net extends NodeJS.EventEmitter {
```
Comment (rhysd) [inline electron/index.d.ts]: Is it correct that `Net` instance extends `EventEmitter`? I couldn't find that in document.

**Review by rhysd — CHANGES_REQUESTED** [review]

**Comment by rhysd** [comment]
@aliib I'm sorry for catching this late. Please check my comments.

**Comment by aliib** [comment]
ok, thanls @rhysd

File: electron/index.d.ts (hunk 3 above)
Comment (rhysd) [inline electron/index.d.ts]: I got it. I'm not against referring to implementation. Thanks.

**Review by rhysd — APPROVED** [review]

**Comment by rhysd** [comment]
Looks good :+1:

**Pull request merged**